server/src/main/java/com/linecorp/centraldogma/server/internal/api/TokenService.java

```diff
@@ -155,18 +173,25 @@ public CompletableFuture<Token> deleteToken(ServiceRequestContext ctx,
     public CompletableFuture<Token> updateToken(ServiceRequestContext ctx,
                                                 @Param String appId,
                                                 JsonNode node, Author author, User loginUser) {
-        if (node.equals(activation)) {
-            return getTokenOrRespondForbidden(ctx, appId, loginUser).thenCompose(
-                    token -> mds.activateToken(author, appId)
-                                .thenApply(unused -> token.withoutSecret()));
-        }
-        if (node.equals(deactivation)) {
-            return getTokenOrRespondForbidden(ctx, appId, loginUser).thenCompose(
-                    token -> mds.deactivateToken(author, appId)
-                                .thenApply(unused -> token.withoutSecret()));
-        }
-        throw new IllegalArgumentException("Unsupported JSON patch: " + node +
-                                           " (expected: " + activation + " or " + deactivation + ')');
+        return getTokenOrRespondForbidden(ctx, appId, loginUser).thenCompose(
+                token -> {
+                    if (token.isDeleted()) {
+                        throw new IllegalArgumentException(
+                                "The token is deleting. You can not update the token's status");
```

Comment:
- `You can't update the status of the token scheduled for deletion.`
- Could you add the test case for this path?

Reply:
Sure. I added the test `updateToken`

**4 cases** 

- activate -> deactivate
- deactivate -> activate
- not matched `JsonNode`
- delete -> activate